### PR TITLE
Copter: Circle_rate parameter description added

### LINKED ISSUE #21098: Copter: circle mode can't point vehicle at center if CIRCLE_RATE > ATC_SLEW_YAW
As discovered in this [discussion](https://discuss.ardupilot.org/t/strange-behaviour-in-circle-mode/87110/10) if the CIRCLE_RATE is higher than ATC_SLEW_YAW then the vehicle's yaw will not be able to keep pointed at the circle's center.  Most of the time the CIRCLE_RATE is much lower than the ATC_SLEW_YAW so this hasn't been noticed before.

We should probably do one of the following:

1. add a note to the CIRCLE_RATE parameter description <-- I think I like this idea best
2. limit the CIRCLE_RATE to be no more than ATC_SLEW_YAW (I'm not a fan of this idea)
3. print a warning to the user when they enter circle mode for the first time that CIRCLE_RATE is too high
4. add a pre-arm check in the parameter section

### PR DESCRIPTION
**Pull request by prathamesh7255**
closes #21098 
added parameter description in Circle_rate that it must be smaller than Slew_yaw.
I hope I have avoided the mistake I did in my earlier PR asking for a merge commit of my master branch. I would be happy to get feedback if I have done something wrong.

### CODE REVIEW

File: libraries/AC_WPNav/AC_Circle.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ const AP_Param::GroupInfo AC_Circle::var_info[] = {
 
     // @Param: RATE
     // @DisplayName: Circle rate
-    // @Description: Circle mode's turn rate in deg/sec.  Positive to turn clockwise, negative for counter clockwise
+    // @Description: Circle mode's turn rate in deg/sec.  Positive to turn clockwise, negative for counter clockwise. Circle rate must be less than SLEW_YAW parameter
```
Comment (IamPete1) [inline libraries/AC_WPNav/AC_Circle.cpp:22]:
```suggestion
    // @Description: Circle mode's turn rate in deg/sec.  Positive to turn clockwise, negative for counter clockwise. Circle rate must be less than ATC_SLEW_YAW parameter.
```

**Comment by IamPete1** [comment]
Commit should be prefixed with `AC_WPNav: AC_Circle:`, Thanks!

**Comment by prathamesh7255** [comment]
hello everyone, every time when I update my forked repo to sync with this repo with a PR open like this current one, the merge commit of this branch into my master branch is getting added. If anyone knows the reason and solution please let me know.

**Comment by rmackay9** [comment]
@prathamesh7255, could you rebase on master and squash the commits into a single commit?  We normally don't merge "merge branch" commits.  txs!

**Comment by rmackay9** [comment]
@prathamesh7255,

In general it is best to create a new branch for each PR instead of raises the PRs from your master branch.

**Comment by prathamesh7255** [comment]
yeah, it would be much more helpful. thanks

**Pull request closed without merging**